### PR TITLE
Add dashboard to manage orders

### PR DESCRIPTION
**Pull request by MarioRodrigues10**

### CODE REVIEW

**Review by ruioliveira02 — COMMENTED** [review]
image
The underline should appear on Dashboard, not Orders.

image

What is the difference between this and the regular admin orders page?

**Comment by MarioRodrigues10** [comment]
> image The underline should appear on Dashboard, not Orders.
> 
> image
> 
> What is the difference between this and the regular admin orders page?

The purpose of dashboard tab is to save the history of each order. I think it would be better to create a proper table to save that history. What your thoughts on it?

File: lib/store_web/templates/layout/_desktop_navbar.html.heex [hunk 1]
```diff
@@ -33,20 +33,7 @@
...
-                            class: "#{(@current_page == page.key && "") || "hover:text-[#f47c58]"} text-sm font-semibold text-black"
-                          ) %>
-                          <span class={
-                            (@current_page == page.key &&
-                               "bg-black w-4 h-0.5 absolute bottom-5 left-0") || ""
-                          } />
-                        </div>
-                      <% end %>
-                    <% else %>
+                    <%= if assigns[:current_user] && @current_user.role in [:user] do %>
```
Comment (nelsonmestevao) [inline lib/store_web/templates/layout/_desktop_navbar.html.heex:36]:
```suggestion
                    <%= if assigns[:current_user] && @current_user.role == :user do %>
```

File: lib/store_web/router.ex [hunk 2]
```diff
@@ -119,6 +119,8 @@ defmodule StoreWeb.Router do
         live "/orders/:id/show/edit", OrderLive.Edit, :edit
 
         live "/users", UserLive.Index, :index
+
+        live "/dashboard", DashboardLive.Index, :index
```
Comment (nelsonmestevao) [inline lib/store_web/router.ex:123]: It would make more sense to be the first entry

File: lib/store_web/live/backoffice/user_live/index.ex [hunk 3]
```diff
@@ -1,18 +1,17 @@
...
-    {:ok, assign(socket, :orders, Inventory.list_orders() |> Repo.preload(:user))}
+    {:ok, assign(socket, :users, Accounts.list_users())}
   end
 
   @impl true
   def handle_params(_params, _url, socket) do
     {:noreply,
      socket
-     |> assign(:current_page, :cart)}
+     |> assign(:current_page, :users)}
```
Comment (MarioRodrigues10) [inline lib/store_web/live/backoffice/user_live/index.ex:15]: The changes were implemented to fix the styling of the navbar.

File: lib/store_web/live/backoffice/dashboard_live/index.ex [hunk 4]
```diff
@@ -1,14 +1,12 @@
...
   import Store.Inventory
-  alias Store.Repo
-  alias Store.Inventory
   alias Store.Accounts
 
   @impl true
   def mount(_params, _session, socket) do
     {:ok,
-     assign(socket, :orders, Inventory.list_orders_history() |> Repo.preload([:order, :admin]))}
+     assign(socket, :orders, list_orders_history(preloads: [:order, :admin]) |> Enum.reverse())}
```
Comment (MarioRodrigues10) [inline lib/store_web/live/backoffice/dashboard_live/index.ex:9]: This was made to sort by the most recent updated_at.

**Review by ruioliveira02 — APPROVED** [review]

**Pull request merged**